src/measure_theory/integration.lean

```diff
@@ -729,87 +737,100 @@ begin
+
 /-- Monotone convergence theorem -- somtimes called Beppo-Levi convergence.
 
 See `lintegral_supr_directed` for a more general form. -/
 theorem lintegral_supr
   {f : ℕ → α → ennreal} (hf : ∀n, measurable (f n)) (h_mono : monotone f) :
   (∫⁻ a, ⨆n, f n a) = (⨆n, ∫⁻ a, f n a) :=
-let c : nnreal → ennreal := coe in
-let F (a:α) := ⨆n, f n a in
-have hF : measurable F := measurable.supr hf,
-show (∫⁻ a, F a) = (⨆n, ∫⁻ a, f n a),
 begin
-  refine le_antisymm _ _,
-  { rw [lintegral_eq_nnreal],
-    refine supr_le (assume s, supr_le (assume hsf, _)),
-    refine ennreal.le_of_forall_lt_one_mul_lt (assume a ha, _),
-    rcases ennreal.lt_iff_exists_coe.1 ha with ⟨r, rfl, ha⟩,
-    have ha : r < 1 := ennreal.coe_lt_coe.1 ha,
-    let rs := s.map (λa, r * a),
-    have eq_rs : (const α r : α →ₛ ennreal) * map c s = rs.map c,
-    { ext1 a, exact ennreal.coe_mul.symm },
-    have eq : ∀p, (rs.map c) ⁻¹' {p} = (⋃n, (rs.map c) ⁻¹' {p} ∩ {a | p ≤ f n a}),
-    { assume p,
-      rw [← inter_Union, ← inter_univ ((map c rs) ⁻¹' {p})] {occs := occurrences.pos [1]},
-      refine set.ext (assume x, and_congr_right $ assume hx, (true_iff _).2 _),
-      by_cases p_eq : p = 0, { simp [p_eq] },
-      simp at hx, subst hx,
-      have : r * s x ≠ 0, { rwa [(≠), ← ennreal.coe_eq_zero] },
-      have : s x ≠ 0, { refine mt _ this, assume h, rw [h, mul_zero] },
-      have : (rs.map c) x < ⨆ (n : ℕ), f n x,
-      { refine lt_of_lt_of_le (ennreal.coe_lt_coe.2 (_)) (hsf x),
-        suffices : r * s x < 1 * s x, simpa [rs],
-        exact mul_lt_mul_of_pos_right ha (zero_lt_iff_ne_zero.2 this) },
-      rcases lt_supr_iff.1 this with ⟨i, hi⟩,
-      exact mem_Union.2 ⟨i, le_of_lt hi⟩ },
-    have mono : ∀r:ennreal, monotone (λn, (rs.map c) ⁻¹' {r} ∩ {a | r ≤ f n a}),
-    { assume r i j h,
-      refine inter_subset_inter (subset.refl _) _,
-      assume x hx, exact le_trans hx (h_mono h x) },
-    have h_meas : ∀n, is_measurable {a : α | ⇑(map c rs) a ≤ f n a} :=
-      assume n, is_measurable_le (simple_func.measurable _) (hf n),
-    calc (r:ennreal) * integral (s.map c) = (rs.map c).range.sum (λr, r * volume ((rs.map c) ⁻¹' {r})) :
-        by rw [← const_mul_integral, integral, eq_rs]
-      ... ≤ (rs.map c).range.sum (λr, r * volume (⋃n, (rs.map c) ⁻¹' {r} ∩ {a | r ≤ f n a})) :
-        le_of_eq (finset.sum_congr rfl $ assume x hx, by rw ← eq)
-      ... ≤ (rs.map c).range.sum (λr, (⨆n, r * volume ((rs.map c) ⁻¹' {r} ∩ {a | r ≤ f n a}))) :
-        le_of_eq (finset.sum_congr rfl $ assume x hx,
-          begin
-            rw [volume, measure_Union_eq_supr_nat _ (mono x), ennreal.mul_supr],
-            { assume i,
-              refine is_measurable.inter ((rs.map c).preimage_measurable _) _,
-              refine (hf i).preimage _,
-              exact is_measurable_of_is_closed (is_closed_ge' _) }
-          end)
-      ... ≤ ⨆n, (rs.map c).range.sum (λr, r * volume ((rs.map c) ⁻¹' {r} ∩ {a | r ≤ f n a})) :
+  set c : nnreal → ennreal := coe,
```

Comment:
Could you add some comments to this proof explaining where you are, what you are doing, and what remains to be done, to improve readability? I know there were no comments in the original proof, but since you have essentially completely rewritten it it should not be hard for you. For proofs that are more than 10 lines long I think it helps a lot.

Reply:
Actually, the diff is large beacuase I changed the indentation. If you open it with "ignore whitespace", you'll see that I just moved everything inside `begin ... end` and moved one of the inequalities to a separate lemma.

Reply:
OK. If you didn't look at the details, don't worry about adding comments.